app/controllers/application_controller.rb

```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fweets"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if !logged_in?
+      erb :'/users/create_user'
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/signup' do
+
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+    if user.save
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+
+  end
+
+  get '/login' do
+    if !logged_in?
+      erb :'/users/login'
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @all_tweets = Tweet.all
+      erb :'/tweets/tweets'
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+    end
+    redirect "/login"
+  end
+
+  get '/users/:slug' do
+    #if logged_in?
+      @user = User.find_by_slug(params[:slug])
+      @all_tweet = @user.tweets
+      erb :'/users/show'
+    #else
+    #  redirect "/login"
+    #end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+    if !params[:content].empty?
+      tweet = Tweet.create(content: params[:content])
+      tweet.user = current_user
+      tweet.save
+      redirect "/tweets/#{tweet.id}"
+    else
+      redirect "/tweets/new"
+    end
+  end
+
+  get '/tweets/:id' do
+
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      @user = @tweet.user
+      erb :'/tweets/show_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  delete '/tweets/:id' do
+    tweet = Tweet.find(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.